### PR TITLE
chore: add fuzz tests and daily fuzz workflow

### LINKED ISSUE #835: Fuzzing

### PR DESCRIPTION
**Pull request by mildwonkey**
## Description
Remediating the "fuzzing" vulnerability in the security findings 

**NOTE FOR REVIEWERS** - I also turned off the errcheck linter for tests:
> I got _really annoyed_ at the errcheck linter complaining about tests not checking for errors: all of the fuzz tests are expected to MAYBE produce an error, and tbh I would've turned this linter off for tests sooner if I'd known how. But good news, now I know how to turn off specific linters just for `_test.go` files. If we don't like that, I'll remove the change and add the exceptions to each test, but again I regret not doing this from the start 😂  

I added some fuzz tests, using the very precise (sorry, "precise") heuristic of eyeballing a bunch of files and going with my gut. If a method dealt with bytes or arrays I gave it a closer look. (I also wrote some tests and deleted them later because they were very low value: not every byte is a good candidate for fuzzing). This is less of a "mission complete" PR and more of a "groundwork and examples for future-us" PR. 

Fuzzing tests can be slow and (at least for now) annoying to run: they can't be parallelized, and we can't even run multiple fuzz tests in a single package in a single command - they need to be explicitly run, one at a time, or you get errors like this and need to add `$` to the end of the test name to disambiguate:
 
`testing: will not fuzz, -fuzz matches more than one fuzz test: [FuzzCompareControls FuzzCompareControlsInt]"` 

Based on both of those, I added the fuzzing tests as a daily cron job instead of running them on every PR. I made the length configurable but we should default it to whatever makes most sense in CI. I left it at 10s, which was the maximum I'd tolerate while running from my laptop, but maybe 30-60s is better for the daily? TBH I don't think fuzzing is hugely valuable for this project, because we work with a lot of concrete types and strings (instead of doing a lot of things with really random user input) and also write good tests in general, which is why I wanted this to be effective and useful, but not too intrusive. 

## Related Issue
Relates to #730
Closes #835 

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

### CODE REVIEW

**Comment by brandtkeller** [comment]
Going to invest some more time in this shortly - but overall appreciate the instrumenting of fuzzing

**Comment by meganwolf0** [comment]
I like this! And this is just a general comment about testing, but I think it would be nice to have like a "testing best practices" or something doc, probably in community/contrib that talks briefly about when we expect certain kinds of tests for things (fuzzy ones, etc). IDK if this is a good PR to at least introduce the concept of that for fuzz tests in particular, but yeah just throwing it out there.

File: src/cmd/dev/validate_test.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func FuzzRunSingleValidation(f *testing.F) {
+	for _, tc := range []string{"../../test/unit/common/oscal/valid-component.yaml", "../../test/unit/common/oscal/valid-component-metadata-injected.yaml"} {
```
Comment (meganwolf0) [inline src/cmd/dev/validate_test.go]: Believe the inputs here should be down in `src/test/unit/common/validation/<pick-your-favorite>`

**Review by meganwolf0 — COMMENTED** [review]
All lgtm! Only thing is a suggestion to change the fuzz sample in cmd/dev tests to a shape more like the validation.yaml, since that is the expected input there

File: .github/workflows/test-fuzz.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+name: Unit Testing
+
+permissions:
+  contents: read
+
+on:
+  schedule:
+    cron: 0 0 * * *
```
Comment (meganwolf0) [inline .github/workflows/test-fuzz.yaml]: should this be a list? https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#schedule

Comment (mildwonkey) [reply .github/workflows/test-fuzz.yaml]: yes it should!

**Comment by mildwonkey** [comment]
Hopefully that also fixed the job name not showing up in the workflows tab (it was using the file name)

**Review by meganwolf0 — APPROVED** [review]

**Review by brandtkeller — APPROVED** [review]
Appreciative of this being added! Looks like a great scope for first iteration and don't disagree with your thoughts.

**Pull request merged**